rakelib/form526.rake

```diff
@@ -226,4 +226,127 @@ namespace :form526 do
     end
     puts "reuploaded files for #{form_submissions.count} submissions"
   end
+
+  desc 'Convert SIP data to camel case and fix checkboxes [/export/path.csv, ids]'
+  task :convert_sip_data, [:csv_path] => :environment do |_, args|
+    raise 'No CSV path provided' unless args[:csv_path]
+
+    ids = args.extras || []
+
+    def to_olivebranch_case(val)
+      OliveBranch::Transformations.transform(
+        val,
+        OliveBranch::Transformations.method(:camelize)
+      )
+    end
+
+    def un_camel_va_keys!(hash)
+      json = hash.to_json
+      # rubocop:disable Style/PerlBackrefs
+      # gsub with a block explicitly sets backrefs correctly https://ruby-doc.org/core-2.6.6/String.html#method-i-gsub
+      json.gsub!(OliveBranch::Middleware::VA_KEY_REGEX) do
+        key = $1
+        "#{key.gsub('VA', 'Va')}:"
+      end
+      JSON.parse(json)
+      # rubocop:enable Style/PerlBackrefs
+    end
+
+    # make a hash that has both the original and corrupted versions of the disability name.
+    # eg {"myocardial infarction (mi)" => "myocardial infarction (MI)"}
+    def get_disability_array(form_data_hash)
+      new_conditions = form_data_hash['newDisabilities']&.collect { |d| d.dig('condition') } || []
+      rated_disabilities = form_data_hash['ratedDisabilities']&.collect { |rd| rd['name'] } || []
+      new_conditions + rated_disabilities
+    end
+
+    # downcase and remove everything but letters and numbers
+    def simplify_string(string)
+      string&.downcase&.gsub(/[^a-z0-9]/, '')
+    end
+
+    def get_dis_translation_hash(disability_array)
+      dis_translation_hash = {}
+      disability_array.each do |dis|
+        dis_translation_hash[simplify_string(dis)] = dis
+      end
+      dis_translation_hash
+    end
```

Comment:
i kinda wish this threw an exception on the off chance that two unique disabilities simplified to the same string.
Example:
```
disability_array = ['Disability for 12/3', 'Disability F (or 1, 2, & 3).']

def simplify_string(string)
  string&.downcase&.gsub(/[^a-z0-9]/, '')
end

def get_dis_translation_hash(disability_array)
  dis_translation_hash = {}
  disability_array.each do |dis|
    dis_translation_hash[simplify_string(dis)] = dis
  end
  dis_translation_hash
end

get_dis_translation_hash(disability_array)

# => {"disabilityfor123"=>"Disability F (or 1, 2, & 3)."}
```